### ISSUE TITLE
add support window 

### ISSUE THREAD

**Issue by yelkamel** [ryanheise/just_waveform#19]
Hello, 

Thank you for this plugin.

We create an app of meditation and sound relaxing named Evolum (4.9 stars in stores).
We wanted to add audio wave in our app. 
As you know we can't get the audiowave file on the file directly in the app when an user want to relax.

We have a dashboard where internal user can upload audio to be available in the app.

This dashboard is web so we can't generate the audiowave file so we wanted to do by deploying our dashboard in macOS but some of our internal user have only window...

We try to doc something with Google Cloud function when a file is uploaded in our storage with that

https://github.com/bbc/audiowaveform

but we didn't successed...

So if adding window support is not a big challenge maybe that can be usefull for us.

Thanks,

**Comment by ryanheise** [comment]
As I don't personally have Windows, that would have to be a contribution if anyone is willing (although first I'd have to move to the federated plugin model).

But for your use case, I'd still recommend running audiowaveform server-side. It's simple enough to do (or you could just outsource the work.)

**Comment by yelkamel** [comment]
Okay thank you 

if anyone is interessed to do that, contact me.

**Comment by zjjt** [comment]
@yelkamel , Salut , i have a need to run this plugin on windows , how should this be done ?

**Comment by ryanheise** [comment]
Unfortunately I don't have any sample code, and everybody will likely be using a different language in the server side, not necessarily Dart. But it should be simple for anyone who knows how to build a restful service, so if you don't know how to do that, you should first learn how to do that in whatever language you're using on the server side.

**Comment by zjjt** [comment]
@ryanheise I know how to build one in fact my app has its backend written in rust and I thought about creating an endpoint to get the waveform data of an audio that I send to it ...what I am lacking is how to call the program audiowaveform ...I also thought of using the java code you made for the plug in on android to implement the c++ code for Windows using chatGPT since its been so long and my due date is next week and I have no time left to learn c++ ...I am that desperate

**Comment by ryanheise** [comment]
> what I am lacking is how to call the program audiowaveform

Do you mean how to call `exec` to execute a program on the OS? I would expect that to be simple also, so I'm not sure whether that's what you mean is your obstacle. If you mean which command line options to pass to the program, the README of that program describes the options and this library is closely modelled on that.

If you are interested in porting to Windows, you might also want to look at the iOS implementation which is written in C (well, Objective C, but it's still mostly C).

**Comment by zjjt** [comment]
@ryanheise yeah indeed it was exec on the Os ill create that endpoint today. I'll use it as last resort and I'll also try porting It to windows directly thanks for the time I'll look into the ios c code .

**Comment by ryanheise** [comment]
Yes, I have intentionally used that format in order to make it compatible with audiowaveform. Note that audiowaveform has an option to output in JSON as well which just_waveform doesn't support, so just make sure you get it to output in the binary format.

**Comment by zjjt** [comment]
@ryanheise All right understood. I am almost done writing the endpoint. thanks a lot for your time and inputs
